dnd-troops-backend/src/main/java/cz/muni/fi/pa165/w2018/dndtroops/backend/entity/User.java

```diff
@@ -0,0 +1,101 @@
+package cz.muni.fi.pa165.w2018.dndtroops.backend.entity;
+
+import cz.muni.fi.pa165.w2018.dndtroops.backend.enums.UserType;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.util.Objects;
+
+/**
+ * User entity
+ *
+ * Marek Valko
+ */
+@Entity(name = "users")
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotNull
+    @Column(unique = true)
+    private String login;
+
+    @NotNull
+    private String passwordHash;
+
+    @OneToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "hero_id")
+    private Hero hero;
+
+    @NotNull
+    @Enumerated
+    private UserType type;
+
+    @Override
```

Comment:
Move methods to be consistent with rest of the entities: constructor, getters/setters, is methods, equals, hashcode, to string